frontend/javascripts/navbar.js

```diff
@@ -268,7 +270,17 @@ function AnonymousAvatar() {
   return (
     <Popover
       placement="bottomRight"
-      content={<LoginForm layout="horizontal" style={{ maxWidth: 500 }} />}
+      content={
+        <LoginForm
+          layout="horizontal"
+          style={{ maxWidth: 500 }}
+          onLoggedIn={async () => {
+            // We reload the tracing upon successful login to access the dataset with the newly logged in user.
```

Comment:
I think, this will reload all pages (including dashboard, spotlight etc.), but the reload is only needed for tracing (or view dataset) page. For that reason, the `Model.ensureSavedState` would be a problem, as well.

How about having a saga which listens to changes of `activeUser` in the store and then saves the model and reloads the page?

Reply:
This is not necessary anymore